components/ApiKeyInput.tsx

```diff
@@ -1,27 +1,26 @@
-import { useState, useEffect } from 'react';
+import { useMemo } from 'react';
 
-export default function Component({defaultKey = null, keyName}) {
-    const [apiKey, setApiKey] = useState(() => {
-        const storedKey = window.localStorage.getItem(keyName);
-        return storedKey || defaultKey || '';
-    });
+interface Props {
+    keyName: string;
+    defaultKey?: string;
+    onChange?: (key: string) => void;
+}
 
-    useEffect(() => {
-        window.localStorage.setItem(keyName, apiKey);
-    }, [apiKey, keyName]);
```

Comment:
I removed this along with all Input component state management in favor of the Form tracking all API key changes, so that the form can immediately react to keypresses. I think previous behavior was that the key would be saved in local storage but you needed to refresh the page to re-trigger the form's `getItem` call; this way a refresh is no longer needed.

Note however that the `database` key will no longer be set in local storage. I think that's fine though as I saw no usages of that value.